### DISCUSSION TITLE
Store @click handlers do not work when function is not invoked in attribute

### DISCUSSION THREAD

**Discussion by jackvanderbilt-visma** [alpinejs/alpine#4380] [7. Bugs (archived)] [answered]
We encountered an issue where the DOM does not update as expected when omitting parentheses in the @click attribute.

### Working Example:
The following code works as expected:
```html
<!DOCTYPE html>
<html>
    <head>
        <script src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js" defer></script>
    </head>
    <body>
        <button x-data @click="$store.someStore.toggle()" :class="$store.someStore.show && 'bg-black'">
            <span x-text="$store.someStore.show"></span>
        </button>
        <script>
            document.addEventListener('alpine:init', () => {
                Alpine.store('someStore', {
                    show: false,
                    toggle() {
                        this.show = !this.show;
                        console.log('Toggle | Show: ', this.show);
                    }
                });
            });
        </script>
    </body>
</html>
```

### Failing Example:
In this variation, the DOM does **not** update, even though the `console.log` executes:
```html
<!DOCTYPE html>
<html>
    <head>
        <script src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js" defer></script>
    </head>
    <body>
        <button x-data @click="$store.someStore.toggle" :class="$store.someStore.show && 'bg-black'">
            <span x-text="$store.someStore.show"></span>
        </button>
        <script>
            document.addEventListener('alpine:init', () => {
                Alpine.store('someStore', {
                    show: false,
                    toggle() {
                        this.show = !this.show;
                        console.log('Toggle | Show: ', this.show);
                    }
                });
            });
        </script>
    </body>
</html>
```

### Issue:
The key difference between the two examples is the function invocation in the `@click` handler:
- The working example uses `@click="$store.someStore.toggle()"`.
- The non-working example omits the parentheses: `@click="$store.someStore.toggle"`.

In both cases, `console.log` runs and outputs the correct value of `show`. However, when the parentheses are omitted, the value change of `show` does not trigger a DOM update.

### Expected Behavior:
Based on the documentation [here](https://alpinejs.dev/globals/alpine-store), it is shown that the parentheses `()` should be used for store functions. However, in other sections of the documentation, such as [here](https://alpinejs.dev/directives/data), omitting parentheses appears to work too, causing confusion.

### Conclusion:
This seems like a bug with AlpineJS. Even though the `console.log` runs, the DOM does not react to the `show` value change when the parentheses are omitted. Ideally, the DOM should update regardless of how the function is passed in the @click attribute, or at least provide details in the docs that parentheses are required for reactivity in all cases.

### References:
- [Alpine Store Documentation](https://alpinejs.dev/globals/alpine-store)
- [AlpineJS Directives Documentation](https://alpinejs.dev/directives/data)

**Comment by ekwoka** [comment, marked as answer]
When leaving out `()` the method is run with the current context as the `this` instead of the object the method is on. This is true for Stores as well as just nested objects.

This is a known and expected behavior. We should probably make it more clear in the documentation that with `()` and without is NOT universally equal.
